src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -614,7 +614,8 @@ void ObjectCopyRequest<I>::merge_write_ops() {
 
         m_snapshot_sparse_bufferlist[src_snap_seq].insert(
           object_extent.offset, object_extent.length,
-          {io::SPARSE_EXTENT_STATE_DATA, object_extent.length,\
+          {io::SPARSE_EXTENT_STATE_DATA,
+           static_cast<size_t>(object_extent.length),
```

Comment:
I think we should change SparseExtent::length to uint64_t instead.  Not sure why it's size_t -- ReadExtent::length is uint64_t and related length() methods in the surrounding classes all return uint64_t.

Reply:
@idryomov thanks. that makes much more sense. updated and repushed.